File: Backend-main/src/entities/WreathPriceEntity.ts
import { MemorialDonationWreath } from 'src/memorial/interface/memorial.interface';
import { Entity, Column, PrimaryColumn, UpdateDateColumn } from 'typeorm';

@Entity('wreath_price')
export class WreathPriceEntity {
  @PrimaryColumn({
    type: 'enum',
    enum: MemorialDonationWreath,
    enumName: 'memorial_wreath',
  })
  wreath: MemorialDonationWreath;

  @Column({ type: 'int', name: 'price_in_cents', nullable: false })
  priceInCents: number;

  @UpdateDateColumn({ name: 'updated_at', type: 'timestamptz' })
  updatedAt: Date;
}

File: Backend-main/src/entities/WreathPriceEntity.java
package entities;

import entities.enums.MemorialDonationWreath;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.OffsetDateTime;
import org.hibernate.annotations.UpdateTimestamp;

@Entity
@Table(name = "wreath_price")
public class WreathPriceEntity {
  @Id
  @Column(name = "wreath")
  private MemorialDonationWreath wreath;

  @Column(name = "price_in_cents")
  private Integer priceInCents;

  @UpdateTimestamp
  @Column(name = "updated_at", columnDefinition = "timestamptz")
  private OffsetDateTime updatedAt;

  public MemorialDonationWreath getWreath() {
    return wreath;
  }

  public void setWreath(MemorialDonationWreath wreath) {
    this.wreath = wreath;
  }

  public Integer getPriceInCents() {
    return priceInCents;
  }

  public void setPriceInCents(Integer priceInCents) {
    this.priceInCents = priceInCents;
  }

  public OffsetDateTime getUpdatedAt() {
    return updatedAt;
  }

  public void setUpdatedAt(OffsetDateTime updatedAt) {
    this.updatedAt = updatedAt;
  }
}
